MacrobenchmarkSample/macrobenchmark/src/main/java/com/example/macrobenchmark/clicks/ClickLatencyBenchmark.kt

```diff
@@ -0,0 +1,157 @@
+                setupBlock = {
+                    if (firstStart) {
+                        val intent = Intent("$packageName.RECYCLER_VIEW_ACTIVITY")
+                        startActivityAndWait(intent)
+                        firstStart = false
+                    }
+                    Thread.sleep(500)
+                }
+        ) {
+            clickOnText("Item 0")
+        }
+    }
+
+    @Test
+    fun composeLazyColumnClick() {
+        var firstStart = true
+        benchmarkRule.measureRepeated(
+                packageName = TARGET_PACKAGE,
+                metrics = listOf(TraceSectionMetric("ClickTrace")),
+                compilationMode = CompilationMode.Full(),
+                startupMode = null,
+                iterations = ITERATIONS,
+                setupBlock = {
+                    if (firstStart) {
+                        val intent = Intent("$packageName.COMPOSE_ACTIVITY")
+                        startActivityAndWait(intent)
+                        firstStart = false
+                        Thread.sleep(500)
+                    }
+                }
+        ) {
+            clickOnText("Item 0")
+        }
+    }
+
+    @Test
+    fun listViewClick() {
+        var firstStart = true
+        benchmarkRule.measureRepeated(
+                packageName = TARGET_PACKAGE,
+                metrics = listOf(TraceSectionMetric("ClickTrace")),
+                compilationMode = CompilationMode.Full(),
+                startupMode = null,
+                iterations = ITERATIONS,
+                setupBlock = {
+                    if (firstStart) {
+                        val intent = Intent("$packageName.LIST_VIEW_ACTIVITY")
+                        startActivityAndWait(intent)
+                        firstStart = false
+                        Thread.sleep(500)
+                    }
+                }
+        ) {
+            clickOnText("Item 0")
+        }
+    }
+
+    @Test
+    fun scrollViewClick() {
+        var firstStart = true
+        benchmarkRule.measureRepeated(
+                packageName = TARGET_PACKAGE,
+                metrics = listOf(TraceSectionMetric("ClickTrace")),
+                compilationMode = CompilationMode.Full(),
+                startupMode = null,
+                iterations = ITERATIONS,
+                setupBlock = {
+                    if (firstStart) {
+                        val intent = Intent("$packageName.SCROLL_VIEW_ACTIVITY")
+                        startActivityAndWait(intent)
+                        firstStart = false
+                        Thread.sleep(500)
+                    }
+                }
+        ) {
+            clickOnText("Item 0")
+        }
+    }
+
+    private fun MacrobenchmarkScope.clickOnText(text: String) {
+        device
+                .findObject(By.text(text))
+                .click()
+        // Chill to ensure we capture the end of the click span in the trace.
+        Thread.sleep(100)
```

Comment:
Consider using `waitForIdle(...)` instead; here and elsewhere.

Reply:
@pyricau please apply this suggestion and replace the sleeps.